src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2205,6 +2301,33 @@ private Symbol AsMemberOfResultType(Symbol symbol)
             return symbol;
         }
 
+        private static Symbol AsMemberOfTupleType(TupleTypeSymbol tupleType, Symbol symbol)
+        {
+            if (symbol.ContainingType.Equals(tupleType, TypeCompareKind.CompareNullableModifiersForReferenceTypes))
+            {
+                return symbol;
+            }
+            var underlyingType = tupleType.UnderlyingNamedType;
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Field:
+                    {
+                        var index = ((FieldSymbol)symbol).TupleElementIndex;
+                        if (index >= 0)
+                        {
+                            return tupleType.TupleElements[index];
+                        }
+                        return new TupleFieldSymbol(tupleType, (FieldSymbol)AsMemberOfType(underlyingType, ((TupleFieldSymbol)symbol).UnderlyingField), index);
```

Comment:
>return new TupleFieldSymbol(tupleType, (FieldSymbol)AsMemberOfType(underlyingType, ((TupleFieldSymbol)symbol).UnderlyingField), index); [](start = 24, length = 135)

This is wrong and I believe I already commented on this approach in another PR. Management of members should be done by TupleTypeSymbol. See TupleTypeSymbol.GetTupleMemberSymbolForUnderlyingMember API for example. If that doesn't work, I'd like to get a better understanding why and then we can decide what to do about that.